common:
  # 4개의 카메라 토픽을 리스트로 정의
  
  img_topic: "/alphasense_driver_ros/cam0" # ONLY_IF num_of_cam == 1 
  img_topics: 
    - "/alphasense_driver_ros/cam0" 
    - "/alphasense_driver_ros/cam1" 
    # - "/alphasense_driver_ros/cam3" 
    # - "/alphasense_driver_ros/cam4" 
  lid_topic: "/os_cloud_node/points"
  imu_topic: "/alphasense_driver_ros/imu"
  num_of_cam: 2 # 카메라 개수 명시
  img_en: 1
  lidar_en: 1
  ros_driver_bug_fix: false
  sync_time_window: 0.05
  minimum_simultaneous_frame_num: 1
  lidar_window_size: 0.1
  en_sliding_window_ICP: false
  en_cam_backprop: true
  mapping_after_vio: false

extrin_calib:
  # LiDAR to IMU Extrinsics (T_il)
  extrinsic_T: [-0.037, -0.008, -0.026]
  extrinsic_R: [1, 0, 0, 0, 1, 0, 0, 0, 1]

  R_cl_0: [-0.001070287110747692, 0.9999775099474943, -0.006620731433109983, 
    -0.007807037138439998, 0.006612177862667906, 0.9999476632679489, 
    0.999968951846316, 0.001121919391599835, 0.007799784629024531]
  P_cl_0: [-0.05475654, -0.01627072, -0.08341641]

  R_cl_1: [0.009858324983608086, 0.9999487481397873, 0.002305324747244264, 
    -0.0009990649217994085, -0.002295586068900013, 0.9999968660720306, 
    0.9999509064438115, -0.009860597257416004, 0.0009763830843686483]
  P_cl_1: [ 0.0553556,  -0.01590483, -0.08328761]

  R_cl_2: [-0.9999770395874403, 0.0008535162098522019, 0.006722485256069301,
    0.006713129004657366, -0.010528556524205824, 0.9999220386592563,
    0.0009242277345858016, 0.9999442089474772, 0.010522585019780083]
  P_cl_2: [ 0.0310896,  -0.01375924, -0.08075841]

  R_cl_3: [0.9999608005152463, -0.00764705302431521, -0.004463184171536622, 
    0.004463831033452346, 6.752387667768867e-05, 0.9999900347768615, 
    -0.007646675448227889, -0.9999707585827533, 0.00010165638238096575]
  P_cl_3: [-0.03038673, -0.0146311,  -0.09841961]

# [Deprecated]

  # extrinsic_R: [1, 0, 0, 0, 1, 0, 0, 0, 1]
  # R_cl_0: [ 0.006708021451800439,  0.9999264200621176,  -0.01010727015365992,
  #          0.002425643641803532,  0.010091197021688258,  0.9999461405473755,
  #           0.9999745590269407, -0.00673217679702115,  -0.0023577731969991467]
  # P_cl_0: [-0.04586422589354697, 0.012631813183337478, -0.05098782892861867]
  # R_cl_1: [ 0.0016556126470597954,  0.9999840642813064,  -0.005397233569448934,
  #           0.0009350089535379302,  0.005395690615039778,  0.9999850060281122,
  #           0.9999981923508761, -0.0016606342845620903,  -0.0009260608798763448]
  # R_cl_2: [ 0.9999897552434941,  0.0042340516721277015, -0.0016006918802386892,
  #           0.004238470700886935,  -0.9999871881804255,  0.0027674611333545654,
  #           -0.0015889537990238962, -0.0027742172670246527,  -0.9999948894591316]
  # R_cl_3: [-0.9998916894135631,  0.007396041673023157,  0.01272430781071447,
  #           0.01270714046408309,  -0.002363530935315569, 0.9999164676625464,
  #           0.007425498159515405,  0.9999698556902047,  0.002269292399476633]
  # R_cl_4: [0.9999880402484476, -0.000942427662931895, -0.004799082221662863,
  #   0.004797008865106434, -0.0021900897852221157,0.9999860960096802, 
  #   -0.0009529249803788974, -0.9999971576643774, -0.0021855427584387965]



time_offset:
  imu_time_offset: 0.0
  img_time_offset: 0.0018
  exposure_time_init: 0.0

preprocess: 
  point_filter_num: 2
  filter_size_surf: 0.1
  lidar_type: 3 # Ouster
  scan_line: 128
  blind: 0.5 
 
vio:
  max_iterations: 5
  outlier_threshold: 500
  img_point_cov: 100
  patch_size: 4
  patch_pyrimid_level: 4
  normal_en: true
  raycast_en: false
  inverse_composition_en: false
  exposure_estimate_en: true
  inv_expo_cov: 0.1

  dudv_thres: 100.0
  min_cov_pixel: 100
  shiTomasiScore_threshold: 0.0
  min_depth_threshold: 0.0
  max_depth_threshold: 100.0
  min_visual_points: 0
  dismiss_non_outofbound_pixels_from_ref_patch: true
  en_error_se3_backprop: true
  en_pose_linear_interpolate_backprop: false

imu:
  imu_en: true
  imu_int_frame: 30
  acc_cov: 0.5 # 0.1
  gyr_cov: 0.01
  b_acc_cov: 0.0001 # 0.1
  b_gyr_cov: 0.0001 # 0.1

lio:
  max_iterations: 5
  dept_err: 0.02
  beam_err: 0.05
  min_eigen_value: 0.0001 # 0.0025
  voxel_size: 0.4
  max_layer: 2
  max_points_num: 100
  layer_init_num: [5, 5, 5, 5, 5]

local_map:  
  map_sliding_en: false
  half_map_size: 100
  sliding_thresh: 8

uav:
  imu_rate_odom: false
  gravity_align_en: false

publish:
  dense_map_en: false
  pub_effect_point_en: false 
  pub_plane_en: false
  pub_scan_num: 1
  blind_rgb_points: 0.0

evo:
  seq_name: "exp09_cupola"
  pose_output_en: true

pcd_save:
  pcd_save_en: false
  colmap_output_en: false # need to set interval = -1
  filter_size_pcd: 0.15
  interval: -1
  # how many LiDAR frames saved in each pcd file;
  # -1 : all frames will be saved in ONE pcd file, may lead to memory crash when having too much frames.
